packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -76,24 +76,48 @@ class WebDavClient {
           .where((final part) => part.isNotEmpty)
           .join('/');
 
-  /// returns the WebDAV capabilities of the server
-  Future<WebDavStatus> status() async {
+  Future<WebDavMultistatus> _parseResponse(final HttpClientResponse response) async =>
+      WebDavMultistatus.fromXmlElement(xml.XmlDocument.parse(await response.body).rootElement);
+
+  Map<String, String>? _getUploadHeaders({
+    required final DateTime? lastModified,
+    required final DateTime? created,
+    required final int? contentLength,
+  }) {
+    final headers = <String, String>{
+      if (lastModified != null) ...{
+        'X-OC-Mtime': (lastModified.millisecondsSinceEpoch ~/ 1000).toString(),
+      },
+      if (created != null) ...{
+        'X-OC-CTime': (created.millisecondsSinceEpoch ~/ 1000).toString(),
+      },
+      if (contentLength != null) ...{
+        'Content-Length': contentLength.toString(),
+      },
+    };
+    return headers.isNotEmpty ? headers : null;
+  }
+
+  /// Gets the WebDAV capabilities of the server.
+  Future<WebDavOptions> options() async {
     final response = await _send(
       'OPTIONS',
       _constructPath(),
       [200],
     );
     final davCapabilities = response.headers['dav']?.cast<String>().first ?? '';
     final davSearchCapabilities = response.headers['dasl']?.cast<String>().first ?? '';
-    return WebDavStatus(
+    return WebDavOptions(
       davCapabilities.split(',').map((final e) => e.trim()).where((final e) => e.isNotEmpty).toSet(),
       davSearchCapabilities.split(',').map((final e) => e.trim()).where((final e) => e.isNotEmpty).toSet(),
     );
   }
 
-  /// make a dir with [remotePath] under current dir
-  Future<HttpClientResponse> mkdir(
-    final String remotePath, {
+  /// Creates a collection at [path].
+  ///
+  /// See http://www.webdav.org/specs/rfc2518.html#METHOD_MKCOL for more information.
+  Future<HttpClientResponse> mkcol(
+    final String path, {
     final bool safe = true,
```

Comment:
Safe needs to be documented

Reply:
Honestly I would just remove it. We never use it and it feels odd to have such an option. The application itself should decide how it handles the failure

Reply:
In the future we should just create a few exceptions that model the different status codes, then it will be much easier for applications to handle specific errors.